python/astro_metadata_translator/translator.py

```diff
@@ -255,26 +307,49 @@ def __init_subclass__(cls, **kwargs):
         if hasattr(cls, "name") and cls.name is not None:
             MetadataTranslator.translators[cls.name] = cls
 
+        # Check that we have not inherited constant/trivial mappings from
+        # parent class that we have already applied. Empty maps are always
+        # assumed okay
+        const_map = cls._const_map if cls._const_map and cls.defined_in_class("_const_map") else {}
+        trivial_map = cls._trivial_map if cls._trivial_map and cls.defined_in_class("_trivial_map") else {}
+
+        # Check for shadowing
+        trivials = set(trivial_map.keys())
+        constants = set(const_map.keys())
+        both = trivials & constants
+        if both:
+            log.warning("%s: defined in both const_map and trivial_map: %s",
+                        cls.__name__, ", ".join(both))
+
+        for name in both:
+            if cls.defined_in_class(f"to_{name}"):
+                # Must be trivial or constant and constant overrides
+                location = "by _trivial_map"
+                if name in constants:
+                    location = "by _const_map"
+                log.warning("%s: %s is defined explicitly but will be replaced %s",
+                            cls.__name__, name, location)
```

Comment:
No f-string here? :)

Reply:
log messages are meant to defer string formatting until it is known that the message will be issued.

Reply:
fair enough